### PR TITLE
chore: run CI as non-root user

### PR DESCRIPTION
**Pull request by DominicKramer**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -118,19 +123,26 @@ jobs:
           command: |
             npm install
             npm link
+          environment:
+            NPM_CONFIG_PREFIX: /home/node/.npm-global
```
Comment (jinwoo) [inline .circleci/config.yml]: Why is this needed?

Comment (DominicKramer) [reply .circleci/config.yml]: I've created this CI based on the changes to the logging-winston library, and the that library has the `NPM_CONFIG_PREFIX` environment variable  set in various places.

Comment (alexander-fenster) [reply .circleci/config.yml]: `npm link` will try to change system directory (somewhere in `/usr` and it will fail on Circle when running as non-root, hence we set `NPM_CONFIG_PREFIX`).

File: .circleci/config.yml [hunk 2]
```diff
@@ -204,6 +223,7 @@ jobs:
   publish_npm:
```
Comment (alexander-fenster) [inline .circleci/config.yml]: `working_directory` should be updated here as well

Comment (alexander-fenster) [reply .circleci/config.yml]: I'm pointing to the line __above__ the `publish_npm` line - it still has `/var/`

File: package.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
   "scripts": {
     "docs": "node node_modules/.bin/repo-tools exec -- jsdoc -c .jsdoc.js",
     "generate-scaffolding": "node node_modules/.bin/repo-tools generate all && node node_modules/.bin/repo-tools generate lib_samples_readme -l samples/ --config ../.cloud-repo-tools.json",
-    "lint": "node node_modules/.bin/repo-tools lint --cmd eslint -- build/src/ samples/ build/system-test/ build/test/",
+    "lint": "node node_modules/.bin/repo-tools lint --cmd eslint -- src/ samples/ system-test/ test/",
```
Comment (jinwoo) [inline package.json]: We have .js files only in the samples/ directory, don't we?

Comment (alexander-fenster) [reply package.json]: Does `eslint` support typescript at all? For me it looks like it does not, at least without some plugin. I guess it's not the right way to fix linting...

Comment (DominicKramer) [reply package.json]: My thought was that if we needed to add any Javascript files to the other directories (`src`, `test`, etc.) then they will be linted.  I think this question of how to do linting should be handled is outside the scope of this PR.  I'll make an issue to address it.
